### PR TITLE
cli/command/container: implement `docker run --annotation`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**
Implemented `docker run --annotation`.

For moby/moby#45025 (Docker v24, API v1.43).

`docker run --annotation foo=bar` is similar to `podman run --annotation foo=bar`, however, unlike Podman, Docker implementation also accepts an annotation with an empty value. (`docker run --annotation foo`)

**- How I did it**

Added the `--annotation` flag as an `*opts.MapOpts`.

**- How to verify it**

Run `docker run --annotation foo=bar`, and run `docker inspect` to confirm that the annotation is set.

**- Description for the changelog**

cli/command/container: implement `docker run --annotation`

**- A picture of a cute animal (not mandatory but encouraged)**
🐧

### CODE REVIEW

**Review by neersighted — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
> Thanks! This should _probably_ be fine, but before we merge, I want to do a quick check look; ISTR we discussed some of this, and whether or not to add this "as-is", have a "porcelain" option (for runtime annotations), etc. (But it's been some time, so I need to read up / read back on some of that).

This feature has been already implemented in `podman run --annotation`, so my opinion is to just follow their CLI design so that their users can easily adopt `docker run --annotation` too

**Review by albers — APPROVED** [review]
Completions LGTM, thanks.

**Comment by AkihiroSuda** [comment]
@thaJeztah Can we merge this?

**Comment by thaJeztah** [comment]
We discussed this in the maintainers meeting, and generally are ok with the current implementation. There were some concerns;

- Is "annotation" confusing (vs "label")? We could probably borrow some of the wording from https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/ to explain the differences
- Is it clear enough this is passed through to the runtime? We discussed describing it as "stored on the container, and passed through to the OCI runtime" 
- We discussed the `=` being optional; we don't expect these to (like `--env`) ever take values from the environment, and that to be ambiguous, so we're good with that (easier to use)
- We _do_ want @Oyabun to have a quick look before merging.

Thanks!

File: cli/command/container/opts.go
Comment (tianon) [inline cli/command/container/opts.go:303]: Something like this, maybe?

```suggestion
	flags.Var(copts.annotations, "annotation", "Add annotations to the container (passed through to the OCI runtime)")
```

(for the relevant item in https://github.com/docker/cli/pull/4156#issuecomment-1507468378)

Comment (thaJeztah) [reply cli/command/container/opts.go:303]: ^^ LGTM (forgot to write that down, but we discussed that as description for the flag) (thanks!)

File: contrib/completion/fish/docker.fish
Comment (thaJeztah) [inline contrib/completion/fish/docker.fish:181]: (don't forget to update these as well when updating the flag description) ❤️

File: cli/command/container/opts.go
Comment (AkihiroSuda) [reply cli/command/container/opts.go:303]: done

File: contrib/completion/fish/docker.fish
Comment (AkihiroSuda) [reply contrib/completion/fish/docker.fish]: done

**Review by tianon — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

I haven't seen strong voices about alternative naming for the flag, so I'll bring this one in. Thanks!

**Pull request merged**
